pkg/controller/util.go

```diff
@@ -36,6 +38,25 @@ func (c *Controller) pvcFromKey(key interface{}) (*v1.PersistentVolumeClaim, err
 	return pvc, nil
 }
 
+func pvcFromKey2(informer cache.SharedIndexInformer, key interface{}) (*v1.PersistentVolumeClaim, error) {
+	keyString, ok := key.(string)
+	if !ok {
+		return nil, errors.New("key object not of type string")
+	}
+	obj, ok, err := informer.GetIndexer().GetByKey(keyString)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error getting key %q from cache", keyString)
+	}
+	if !ok {
+		return nil, errors.Errorf("key %q not found in cache", keyString)
+	}
+	pvc, ok := obj.(*v1.PersistentVolumeClaim)
```

Comment:
this func and podFromKey2 are the same down to this point in each.  could probably reduce some duplicate code